metadata/metadata.go

```diff
@@ -824,14 +858,49 @@ func (this *modelResource) Notify(lookup ResourceLookup, op operation, that Reso
 }
 
 func (resource *modelResource) UpdateStatus(status pb.ResourceStatus) error {
-	resource.serialized.Status = &status
 	return nil
 }
 
 func (resource *modelResource) UpdateSchedule(schedule string) error {
 	return fmt.Errorf("not implemented")
 }
 
+func (resource *modelResource) Update(lookup ResourceLookup, updateRes Resource) error {
+	deserialized := updateRes.Proto()
+	updateModel, ok := deserialized.(*pb.Model)
+	if !ok {
+		return errors.New("failed to deserialize existing model record")
+	}
+	updated := unionNameVariants(resource.serialized.Features, updateModel.Features)
+	resource.serialized.Features = updated
+	updated = unionNameVariants(resource.serialized.Trainingsets, updateModel.Trainingsets)
+	resource.serialized.Trainingsets = updated
+	return nil
+}
+
+func unionNameVariants(source, destination []*pb.NameVariant) []*pb.NameVariant {
+	type nameVariantKey struct {
+		Name    string
+		Variant string
+	}
+
+	set := make(map[nameVariantKey]bool)
+
+	for _, nameVariant := range destination {
+		key := nameVariantKey{Name: nameVariant.Name, Variant: nameVariant.Variant}
```

Comment:
you can write this as:
key := nameVariantKey{nameVariant.Name, nameVariant.Variant}
more typical in Go when you have a simple struct like this and fields are implied